src/app/main/statistics/queries/by-type/by-type.component.ts

```diff
@@ -1,12 +1,48 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 import { Statistics } from "../../../../shared/models/order.model";
+import { TypeGuery } from 'src/app/shared/models/type-query.model';
+import { OrderItemDTO } from 'src/app/shared/models/order-item.model';
 
 @Component({
   selector: 'app-by-type',
   templateUrl: './by-type.component.html',
   styleUrls: ['./by-type.component.scss']
 })
-export class ByTypeComponent {
+export class ByTypeComponent implements OnInit {
   @Input()
-  statistics : Statistics[] = []
+  statistics: Statistics[] = [];
+  statisticsList: TypeGuery[] = [];
+  types: string[] = [];
+  items: OrderItemDTO[] = [];
+
+  ngOnInit(): void {
```

Comment:
On init content is must to moving into other function. maybe render and call this if update button pushed call render, and we need call render function in on init, because If you are in payment or type query component and change date, data is not be. Angular change detection do not call filtering if statistics input file changed. We need discuss it.